### PR TITLE
bpo-34829: Add missing selection_ methods to the Tkinter Spinbox

### PR DESCRIPTION
**Pull request by j4321**
Add methods selection_from, selection_range, selection_present and selection_to to the Tkinter Spinbox
for consistency with the Entry widget.

https://bugs.python.org/issue34829

### CODE REVIEW

**Review by serhiy-storchaka — CHANGES_REQUESTED** [review]
Please add tests, a news entry, a What's New entry, and add your name in Misc/ACKS.

File: Doc/whatsnew/3.8.rst [hunk 1]
```diff
@@ -167,6 +167,15 @@ Added :attr:`SSLContext.post_handshake_auth` to enable and
 post-handshake authentication.
 (Contributed by Christian Heimes in :issue:`34670`.)
 
+tkinter
+-------
+
+Added methods :meth:`~tkinter.Spinbox.selection_from`,
+:meth:`~tkinter.Spinbox.selection_present`,
+:meth:`~tkinter.Spinbox.selection_range` and 
+:meth:`~tkinter.Spinbox.selection_to`,
```
Comment (serhiy-storchaka) [inline Doc/whatsnew/3.8.rst]: The last comma is redundant.

Please add also `(Contributed by ...)`

File: Misc/NEWS.d/next/Library/2018-10-08-16-04-36.bpo-34829.B7v7D0.rst [hunk 2]
```diff
@@ -0,0 +1,3 @@
+Add methods ``selection_from``, ``selection_range``, ``selection_present``
+and ``selection_to`` to the ``tkinter.Spinbox`` for consistency with the
+``tkinter.Entry`` widget.
```
Comment (serhiy-storchaka) [inline Misc/NEWS.d/next/Library/2018-10-08-16-04-36.bpo-34829.B7v7D0.rst]: Please add `Patch by ...`.

**Review by serhiy-storchaka — COMMENTED** [review]
Thank you @j4321. Would be nice to add also tests for existing selection methods and add the same or similar tests for the Entry widget.

**Comment by serhiy-storchaka** [comment]
Seems there are whitespace issues in rst files.

**Comment by j4321** [comment]
@serhiy-storchaka ``Spinbox.selection_element()`` raises ``TclError: expected integer but got "none"`` while it should return the currently selected element according to the docstring. I think I have identified the origin of this issue in  ``Spinbox.selection``. Should I fix it in this pull request or should I open separate issue?

**Comment by serhiy-storchaka** [comment]
This looks like a bug. Please open a separate issue.

File: Doc/whatsnew/3.8.rst [hunk 3]
```diff
@@ -172,9 +172,9 @@ tkinter
 
 Added methods :meth:`~tkinter.Spinbox.selection_from`,
 :meth:`~tkinter.Spinbox.selection_present`,
-:meth:`~tkinter.Spinbox.selection_range` and 
-:meth:`~tkinter.Spinbox.selection_to`,
-in the :class:`tkinter.Spinbox` class.
+:meth:`~tkinter.Spinbox.selection_range` and
+:meth:`~tkinter.Spinbox.selection_to`
+in the :class:`tkinter.Spinbox` class (Contributed by Juliette Monsel).
```
Comment (serhiy-storchaka) [inline Doc/whatsnew/3.8.rst]: Add a reference to the issue. Take other entries as a pattern.

File: Doc/whatsnew/3.8.rst.bak [hunk 4]
```diff
@@ -0,0 +1,401 @@
+****************************
```
Comment (serhiy-storchaka) [inline Doc/whatsnew/3.8.rst.bak]: Remove this file.

File: Misc/ACKS [hunk 5]
```diff
@@ -1837,3 +1837,4 @@ Jelle Zijlstra
 Gennadiy Zlobin
 Doug Zongker
 Peter Åstrand
+Juliette Monsel
```
Comment (serhiy-storchaka) [inline Misc/ACKS]: And the last. Names in this list are in rough alphabetical order by last names. Move your name to the correct place.

**Review by serhiy-storchaka — APPROVED** [review]

File: Misc/ACKS (hunk 5 above)
Comment (j4321) [reply Misc/ACKS]: Sorry for all the mistakes and thanks for taking the time to point them out, I'll try to do everything right from the start next time.

**Comment by serhiy-storchaka** [comment]
Thank you for your contribution Juliette! Congratulate with your first commit to CPython!

**Pull request merged**